modules/correlation/groupingby.c

```diff
@@ -198,6 +203,11 @@ _clone(LogPipe *s)
   GroupingBy *self = (GroupingBy *) s;
   GroupingBy *cloned;
 
+  if (self->next_id == NULL)
+    {
+      self->next_id = g_atomic_rc_box_new0(guint);
```

Comment:
Our GLib min version is 2.32: https://github.com/syslog-ng/syslog-ng/blob/master/configure.ac#L43
`atomic_rc_box` is available from 2.58: https://docs.gtk.org/glib/func.atomic_rc_box_new.html

We should try to find something that does not require raising our GLib min version.

As far as I remember cloning happens during config initialization in the main thread so we don't need to account for thread safety.

Reply:
`lib/atomic.h` and `lib/atomic-gssize.h` can be used, if needed; but I also believe that this does not have to be thread safe as we clone in the main thread during initialization.

Reply:
> Our GLib min version is 2.32: https://github.com/syslog-ng/syslog-ng/blob/master/configure.ac#L43 `atomic_rc_box` is available from 2.58: https://docs.gtk.org/glib/func.atomic_rc_box_new.html
> 
> We should try to find something that does not require raising our GLib min version.
> 
> As far as I remember cloning happens during config initialization in the main thread so we don't need to account for thread safety.

I have changed the implementation to a simple self-made one that does not rely on glib and also is not tgread safe.

If there's a better way to do this please let me know.